src/why3prop.ml

```diff
@@ -0,0 +1,122 @@
+(** Translation of lambdapi goals to why3 goals in propositional case *)
+
+open Terms
+open Extra
+
+exception NoGoalTranslation
+
+(* a type to present the list of why3 constants and lambdapi terms *)
+type cnst_table = (term * Why3.Term.lsymbol) list
+
+(* number of axioms proved whith the Why3 tactic *)
+let nbr_axioms : int ref = ref 0
+
+(** [get_newname ()] generates a new axiom name. *)
+let get_newname () : string =
+    nbr_axioms := !nbr_axioms + 1;
+    "Why3axiom_" ^ (string_of_int !nbr_axioms)
+
+(* builtins configuration for propositional logic *)
+type prop_config =
+  { symb_P     : sym * pp_hint (** Encoding of propositions.        *)
+  ; symb_T     : sym * pp_hint (** Encoding of types.               *)
+  ; symb_or    : sym * pp_hint (** Disjunction(∨) symbol.           *)
+  ; symb_and   : sym * pp_hint (** Conjunction(∧) symbol.           *)
+  ; symb_imp   : sym * pp_hint (** Implication(⇒) symbol.           *)
+  ; symb_bot   : sym * pp_hint (** Bot(⊥) symbol.                   *)
+  ; symb_not   : sym * pp_hint (** Not(¬) symbol.                   *) }
+
+(** [get_prop_config builtins] set the builtins configuration using
+    [builtins] *)
+let get_prop_config : Proof.builtins -> prop_config = fun builtins ->
+  let find_sym key =
+    try StrMap.find key builtins with Not_found ->
+     Console.fatal_no_pos "Builtin symbol [%s] undefined." key
+  in
+  { symb_P     = find_sym "P"
+  ; symb_T     = find_sym "T"
+  ; symb_or    = find_sym "or"
+  ; symb_and   = find_sym "and"
+  ; symb_imp   = find_sym "imp"
+  ; symb_bot   = find_sym "bot"
+  ; symb_not   = find_sym "not" }
+
+(** [translate builtins (hs, g)] translate from lambdapi to Why3 goal [g]
+    using the hypothesis [hs]. *)
+let rec translate : Proof.builtins -> (Env.env * term) ->
```

Comment:
See https://github.com/Deducteam/lambdapi/pull/171/files#r264757477.

Reply:
Please add a more detailed comment on what are the different components of the output.

Reply:
Done in c363bd5e1842c2985e8afee8ce8a2f4201468562.